modules/layers/src/geojson-layer/geojson-layer.js

```diff
@@ -294,48 +382,142 @@ export default class GeoJsonLayer extends CompositeLayer {
         layerProps.lines
       );
 
+    let pointLayerProps;
+    let pointLayerUpdateTriggers;
+    if (pointType === 'circle') {
```

Comment:
- Worth splitting into its own function
- Use `switch`

Reply:
I've changed to `switch` but haven't broken up into functions as it would mean passing through a huge number of variables and would be inconsistent with how the polygon/lines layers above are constructed